File: Introductory Problems/3. repetitions/repetitions.js
function repetitions(sequence) {
  let maxCount = 0, maxItem;
  const longest = [];
  for (let i = 0; i < sequence.length; i++) {
    if (i + 1 < sequence.length && sequence[i] == sequence[i + 1])
      longest.push(sequence[i]);
  }
  for (let codon of longest) {
    maxCount = longest.filter((item) => item == codon).length;
    maxItem = codon;
  }
  return [maxItem, maxCount + 1];
}

const [item, count] = repetitions("AAGGGTCCCCCTGA");
console.log(item, count);

File: Introductory Problems/repetitions/repetitions.py
def repetitions(sequence):
    max_item = ''
    max_count = 0
    longest = []
    for i, codon in enumerate(sequence):
        if i+1 < len(sequence) and sequence[i+1] == codon:
            longest.append(codon)
    for i in longest:
        item_count = longest.count(i)
        if item_count > max_count:
          max_count = item_count
          max_item = i
    return max_item, max_count+1

x,y = repetitions('AAGGGTCCCCCTGA')
print(x, y)
